### PR TITLE
Custom channel testing bugfixes

### LINKED ISSUE #1006: rfq/[bug]: erroneous zero value asset amt TLV due to oracle response
### Background

Not knowing the significance of the flag, a user set the `mockoraclecentpersat` flag to `1`. They then attempted to send a payment, this resulted in a zero value asset amt beign sent: 
```
2024-07-09 23:25:17.201 [DBG] TSVR: ProduceHtlcExtraData called, totalAmount=60975566, htlcBlob=(lnwire.CustomRecords) (len=2) {
2024-07-09 23:25:17.334 [ERR] HSWC: ChannelLink(3596bc22e11333b5c06ea9b4b05ced089acd545df4d1125dca8346b1b5fc64b5:0): failing link: unable to update commitment: unable to fetch aux leaves: unable to generate allocat         ions: unable to prepare output assets: split locator has zero amount with error: internal error
 2024-07-09 23:25:17.201 [DBG] TCHN: Producing HTLC extra data for RFQ ID 9c858135633f1cf295924ef66f7d44d46e54734b49c50524111e72be6b40fba4 (SCID 1233549510114868132): asset ID 1d98fdb40cab6edfd23480403210d8c48a714d9         f693893bb09dde025980df830, asset amount 0
```

### Steps to reproduce

Tell us how to reproduce this issue. Please provide stacktraces and links to code in question.

### Expected behavior

We should do more validation of the values we get from the price oracle. 

### Actual behavior

We use a value without further validation, resulting in a division operation returning `0`, which we use as the outbound HTLC to send.

### LINKED ISSUE #1007: tapgarden: use higher block confidence threshold for re-org watcher on testnet
**Is your feature request related to a problem? Please describe.**

Today we use the re-org watcher to watch for proofs and update them if a re-org happens (update the merkle proof in the proof, etc). 

The default value is 6 confs for mainnet: 
https://github.com/lightninglabs/taproot-assets/blob/fca9d11f027cdb5b81a36c4a8999c6b010ac0c44/tapcfg/config.go#L110-L112

This is somewhat reasonable for mainnet, but on testnet3, hundred block re-orgs can occur. This has resulted in some posted proofs (that were valid at the time), now rendered as invalid: 
```
2024-07-10 14:22:51.288 [WRN] UNIV: encountered an error whilst syncing with server=(universe.ServerAddr) {
 ID: (int64) 1,
 addrStr: (string) (len=40) "testnet.universe.lightning.finance:10029",
 addr: (net.Addr) <nil>
}
: unable to register proofs: unable to register proofs: unable to verify issuance proofs: unable to verify proof: failed to validate proof block header: block hash and block height mismatch; (height: 2867248, hashAtHeight: 00000000d1e5841c9873c053fafc6972da94549198268d1b9d73ddfa184cb6bf, expectedHash: 0000000075f28b79d00152765fcf6feb4d28b4270f8000fd396b675797826d3a)
```

**Describe the solution you'd like**

We should potentially further increase the default mainnet value, and use a much higher value on testnet. 

**Describe alternatives you've considered**

As is, the testnet universe will keep serving those invalid proofs. We could instead start to spot check proofs that we've accepted, and update them if we detect this case (update the merkle proof).

### LINKED ISSUE #1010: [bug]: rfq process error results in tapd crash
### The Issue

When generating an invoice for an asset held in an asset channel, if the rfq process fails, due to non-responsiveness of the peer, incorrect config, etc., the tapd process crashes and litd needs to be restarted. Doesn’t seem to happen when there is no rfq config and the 
```[litcli] peer [peer pub key] sent back an invalid quote, status: PRICE_ORACLE_QUERY_ERR``` 
error is returned. But it does happen when the process times out and this error is received, 
```[litcli] error adding sell order: rpc error: code = Unknown desc = timeout waiting for response from peer ```

the tapd process then fails with this error visible in the output of $ litcli status
```"error":  "received critical error from sub-server (taproot-assets), shutting down: failed to handle outgoing message: error sending outgoing message to stream handler: unable to send message to peer: rpc error: code = NotFound desc = peer is not connected",```

### My environment

I’m on the v0.13.99-alpha.rc3 release of litd on a Ubuntu machine. 

### Steps to reproduce

Check that everything is up and running. 
$ litcli status

Attempt to create an asset based invoice in a situation where it will fail… other node is down, config is incorrect, etc. 
$ litcli --network=testnet --macaroonpath=~/.lnd/data/chain/bitcoin/testnet/admin.macaroon ln addinvoice --asset_id [asset id] --asset_amount 100

Check on that status of tapd
$ litcli status

### Expected behavior

The invoice creation cli command will return error message but litd and tapd will continue running. 

### Actual behavior

The invoice creation cli command returns an error message and litd stays running, but tapd fails. 

Attached are logs from a tapd crash after a failed invoice creation. 
[lndJulRFQCrash.log](https://github.com/user-attachments/files/16170500/lndJulRFQCrash.log)

### PR DESCRIPTION
**Pull request by guggero**
Fixes #1007.
Fixes #1006.
Fixes #1010.

Also adds a fix for the startup dependency issue between `lnd` and `tapd`: Apparently it's possible that `lnd` wants to force close channels before the RPC server is fully started. That then calls into the aux leaf creator which blocks until `tapd` thinks it's fully started (which in turn depends on a successful connection to `lnd`'s RPC).
By making the aux leaf creator fully stateless and allowing some methods of the aux leaf signer that are stateless to be called early, we should be able to fix this startup dependency issue.

### CODE REVIEW

File: rfq/manager.go [hunk 1]
```diff
@@ -577,12 +595,11 @@ func (m *Manager) mainEventLoop() {
 
 		// Handle subsystem errors.
 		case err := <-m.subsystemErrChan:
-			log.Errorf("Manager main event loop received "+
-				"subsystem error: %v", err)
-
 			// Report the subsystem error to the main server.
-			m.cfg.ErrChan <- fmt.Errorf("encountered RFQ "+
-				"subsystem error: %w", err)
+			m.handleError(
```
Comment (ffranr) [inline rfq/manager.go:599]: comment above this line is no longer always true/accurate, depends on what `m.handleError` does

**Review by ffranr — APPROVED** [review]

File: sample-tapd.conf [hunk 2]
```diff
@@ -320,5 +320,12 @@
...
-; experimental.rfq.mockoraclecentpersat=
+; Mock price oracle static asset units per BTC rate (for example number of USD
+; cents per BTC if one asset unit represents a USD cent); use either this or
+; mockoraclesatsperasset
+; experimental.rfq.mockoracleassetsperbtc=
+
+; Mock price oracle static satoshis per asset unit rate (for example number of
+; satoshis to pay for one USD cent if one asset unit represents a USD cent); use
+; either this or mockoracleassetsperbtc
+; experimental.rfq.mockoraclesatsperasset=
```
[the remark this answers is not in the document]
Comment (guggero) [reply sample-tapd.conf:331]: Good point. Added some additional text.

If a user specifies a fraction, they'll get this error: `could not load config: error parsing flags: invalid argument for flag `--taproot-assets.experimental.rfq.mockoracleassetsperbtc' (expected uint64): strconv.ParseUint: parsing "12.3": invalid syntax`

File: tapchannel/aux_traffic_shaper.go [hunk 3]
```diff
@@ -256,6 +256,17 @@ func (s *AuxTrafficShaper) ProduceHtlcExtraData(totalAmount lnwire.MilliSatoshi,
...
+	// If the number of asset units to send is zero due to integer division
+	// and insufficient asset unit precision vs. satoshis, we cannot send
+	// this payment. This should only happen if the amount to pay is very
+	// small (small satoshi or sub satoshi total value) or the price oracle
+	// has given a very high price for the asset.
+	if numAssetUnits == 0 {
+		return 0, nil, fmt.Errorf("asset unit price (%d mSat per "+
+			"asset unit) too high to represent HTLC value of %v",
+			mSatPerAssetUnit, totalAmount)
+	}
```
[the remark this answers is not in the document]
Comment (guggero) [reply tapchannel/aux_traffic_shaper.go:268]: Hmm, I don't think we should just silently overpay, as that'll definitely be exploited at some point. I really think this falls in the same precision category as with the MPP rounding error, which should be fixed by minting assets with a sufficient number of units per BTC.

**Review by jharveyb — APPROVED** [review]
LGTM 👍🏽 

Nice to cut down on state / possible dependency issues.

File: rfq/oracle.go [hunk 4]
```diff
@@ -368,15 +368,29 @@ var _ PriceOracle = (*RpcPriceOracle)(nil)
...
-	expiryDelay uint64
-	usdPerBTC   uint64
+	expiryDelay  uint64
+	mSatPerAsset lnwire.MilliSatoshi
 }
 
 // NewMockPriceOracle creates a new mock price oracle.
-func NewMockPriceOracle(expiryDelay, usdPerBTC uint64) *MockPriceOracle {
+func NewMockPriceOracle(expiryDelay, assetsPerBTC uint64) *MockPriceOracle {
+	mSatPerAsset := lnwire.NewMSatFromSatoshis(btcutil.SatoshiPerBitcoin) /
```
[the remark this answers is not in the document]
Comment (guggero) [reply rfq/oracle.go:377]: It's never updated in the mock. It's a static value configured in the config or CLI args.

File: tapchannel/aux_leaf_creator.go [hunk 5]
```diff
@@ -25,67 +23,12 @@ const (
...
-	cfg *LeafCreatorConfig
-
-	// ContextGuard provides a wait group and main quit channel that can be
-	// used to create guarded contexts.
-	*fn.ContextGuard
-}
-
-// NewAuxLeafCreator creates a new Taproot Asset auxiliary leaf creator based on
-// the passed config.
-func NewAuxLeafCreator(cfg *LeafCreatorConfig) *AuxLeafCreator {
```
Comment (Roasbeef) [inline tapchannel/aux_leaf_creator.go:48]: FWIW, I think we could've kept the struct structure for ease of use/isolation, in say tests.

**Pull request merged**